README.md

```diff
@@ -447,6 +448,181 @@ you can create your own Configuration(s) and compose them with Config's ++ opera
 
 Then you can build as usual with CONFIG=MyConfig.
 
+## <a name="debug"></a> Debugging with GDB
+
+### 1) Generating the RBB emulator
+
+The objective of this section is to use GNU debugger to debug programs RISC-V programs running on the emulator in the same fashion as in [Spike](https://github.com/riscv/riscv-isa-sim#debugging-with-gdb).
+
+For that we need to build a Remote Bit Bang enabled emulator, this could be any of the listed configurations in `rocket-chip/src/main/scala/system/Configs.scala`. Basically all we need is to extend our configuration with JtagDTMSystem that will deal with program loading and interface with the debugger. In the following example we added this extention to the DefaultConfig one.
+
+    class DefaultConfigRBB extends Config(
+    new WithJtagDTMSystem ++ new WithNBigCores(1) ++ new BaseConfig)
+
+    class QuadCoreConfigRBB extends Config(
+    new WithJtagDTMSystem ++ new WithNBigCores(4) ++ new BaseConfig)
+
+To build the configuration we use the command:
+
+    rocket-chip$ cd emulator
+    emulator$ CONFIG=DefaultConfigRBB make
+
+By default the emulator is generated under the name `emulator-freechips.rocketchip.system-DefaultConfigRBB`.
+
+### 2) Compiling the executable
+
+We suppose that `helloworld` is our program, you can use `crt.S`, `syscalls.c` and the linker script `test.ld` to construct your own program, check examples stated in [riscv-tests](https://github.com/riscv/riscv-tests).
+
+In our case we will use the following example:
+
+```
+char text[] = "Vafgehpgvba frgf jnag gb or serr!";
+
+// Don't use the stack, because sp isn't set up.
+volatile int wait = 1;
+
+int main()
+{
+    while (wait)
+        ;
+
+    // Doesn't actually go on the stack, because there are lots of GPRs.
+    int i = 0;
+    while (text[i]) {
+        char lower = text[i] | 32;
+        if (lower >= 'a' && lower <= 'm')
+            text[i] += 13;
+        else if (lower > 'm' && lower <= 'z')
+            text[i] -= 13;
+        i++;
+    }
+
+    while (!wait)
+        ;
+}
+```
+Do not forget to compile with `-g -Og` flags to provide debugging support just as explained [here](https://github.com/riscv/riscv-isa-sim#debugging-with-gdb).
+
+### 3) Launch the emulator
+
+We can then launch the emulator with
+
+    ./emulator-freechips.rocketchip.system-DefaultConfigRBB +jtag_rbb_enable=1 --rbb-port=9823 helloworld
```

Comment:
There is no real point to the `helloworld` argument to the emulator, because you `load` it later through the debugger. What happens if you leave it off?

Reply:
You should note that if you leave off the `--rbb-port` argument, a random port will be assigned.

Reply:
Actually without the program argument, the emulator simply shows the help message.

Reply:
Default RBB if no -rbb-port note added.

Reply:
> Actually without the program argument, the emulator simply shows the help message.

Good to know. Maybe we should add a note here that it's not going to load your program, or that that argument doesn't matter, or use the term "dummy" ( I believe the latter is what the Debug Regressions actually do). Because it doesn't matter what you put there, since FESVR isn't attached it's not going to load that program.

Reply:
Okay. I prefer to let the `helloworld` there so that new people don't get confused when they don't use RBB or something, but notified the fact that it is loaded from GDB (in contrast with Spike) and that's it's a dummy arg. Thanks for clarification.